app/models/concerns/public_activity_trackable.rb

```diff
@@ -0,0 +1,40 @@
+# Tracks updates of:
+# - common attributes updates (visibility_status, discarded_at)
+# - deletes
+#
+# Required to be implemmented by the model
+# - create_activity (from PublicActivity::Common)
+#
+module PublicActivityTrackable
+  extend ActiveSupport::Concern
+
+  included do
+    include PublicActivity::Common
+
+    around_update :track_updates_activity
+  end
+
+  protected
+
+  def track_updates_activity
+    yield
+
+    return unless errors.blank? && previous_changes.present?
+
+    create_activity activity_key_for_last_update, owner: owner
```

Comment:
If we want to manually create activities (for bulk updates, CSV imports) we could extract from here something like `#create_activity_with_current_changes`:

```ruby
module PublicActivityTrackable
  # ..
  def create_activity_with_current_changes!
    # ..
    create_activity activity_key_for_last_update, owner: owner
  end

  protected

  def track_updates_activity
    yield
    create_activity_with_current_changes!
  end
end
```

then for mentioned cases:

```ruby
company.update_attributes()
company.create_activity_with_current_changes!
```

Reply:
I think we could just start without using `update_all` in batch actions.

Reply:
Just `update` and everything should work.

Reply:
I've done in this way. Looks like it works as it should now I think